src/SFML/Graphics/SpriteBatch.cpp

```diff
@@ -0,0 +1,266 @@
+                start       = next;
+            }
+
+            next++;
+        }
+
+        // Deal with leftovers
+        if (start != m_triangleInfos.size())
+        {
+            sendVertices(&m_vertices[start * 3], (next - start) * 3, lastTexture);
+        }
+    }
+
+    else if (m_batchMode == BatchMode::DepthSort || m_batchMode == BatchMode::TextureSort)
+    {
+        std::vector<std::size_t> indices(m_triangleInfos.size());
+        std::size_t              triangleCount = m_triangleInfos.size();
+
+        for (std::size_t i = 0; i < triangleCount; i++)
+            indices[i] = i;
+
+        if (m_batchMode == BatchMode::TextureSort)
+        {
+            const auto& comp = [&](const std::size_t a, const std::size_t b)
+            { return m_triangleInfos[a].texture < m_triangleInfos[b].texture; };
+
+            std::stable_sort(indices.begin(), indices.end(), comp);
+        }
+
+        else
+        {
+            const auto& comp = [&](const std::size_t a, const std::size_t b)
+            {
+                if (m_triangleInfos[a].depth != m_triangleInfos[b].depth)
+                    return m_triangleInfos[a].depth > m_triangleInfos[b].depth;
+                else
+                    return m_triangleInfos[a].texture < m_triangleInfos[b].texture;
+            };
+
+            std::stable_sort(indices.begin(), indices.end(), comp);
+        }
+
+        // Create the array of sorted vertices
+        // We need them sorted so that we feed chunks to RenderTarget
+        m_sortedVertices.resize(m_vertices.size());
+
+        for (std::size_t i = 0; i < indices.size(); i++)
+        {
+            std::size_t newPos           = i * 3;
+            std::size_t oldPos           = indices[i] * 3;
+            m_sortedVertices[newPos]     = m_vertices[oldPos];
+            m_sortedVertices[newPos + 1] = m_vertices[oldPos + 1];
+            m_sortedVertices[newPos + 2] = m_vertices[oldPos + 2];
+        }
+
+        // Batch based on texture change
+        std::size_t    startIndex  = 0;
+        std::size_t    nextIndex   = 0;
+        const Texture* lastTexture = nullptr;
+
+        while (nextIndex < triangleCount)
+        {
+            const Texture* nextTexture = m_triangleInfos[indices[nextIndex]].texture;
+
+            if (nextTexture != lastTexture)
+            {
+                sendVertices(&m_sortedVertices[startIndex * 3], (nextIndex - startIndex) * 3, lastTexture);
+
+                lastTexture = nextTexture;
+                startIndex  = nextIndex;
+            }
+
+            nextIndex++;
+        }
+
+        // Deal with leftovers
+        if (startIndex != m_triangleInfos.size())
+        {
+            sendVertices(&m_sortedVertices[startIndex * 3],
+                         (nextIndex - startIndex) * 3,
+                         m_triangleInfos[indices[startIndex]].texture);
+        }
+    }
+
+    m_vertices.resize(0);
+    m_triangleInfos.resize(0);
+    m_sortedVertices.resize(0);
```

Comment:
What are you intentions here? Do you need to just empty the containers? Do you also need to deallocate all the elements? Resizing to zero seems like a more complicated way to simply call `.clear()`.

Reply:
Just emptying it. I'll change it to `clear`.